test/unit/lib/api-client.js

```diff
@@ -0,0 +1,325 @@
+import { assert } from 'chai';
+import MockServer from '../../util/mock-server';
+import {
+    changeProtocol,
+    changeHost,
+    changePort,
+    checkIn,
+    checkOutItems,
+    createItem,
+    createModel,
+    deleteItem,
+    getAllItems,
+    getAllModels,
+    searchItem,
+    searchModel,
+    searchStudent
+} from '../../../.dist/lib/api-client';
+
+describe('API Client', () => {
+
+    let mockServer = new MockServer();
+
+    before(() => {
+        changeProtocol('http');
+        changeHost('localhost');
+        changePort(8080);
+        return mockServer.start(8080);
+    });
+
+    beforeEach(() => {
+        mockServer.clearExpectations();
+    });
+
+    after(() => {
+        mockServer.stop();
+    });
+
+    it('checkIn', () => {
+        let response = {
+            status: 'success',
+            data: {
+                itemAddress: 'iGwEZUvfA',
+                modelName: 'Resistor'
+            }
+        };
+        mockServer.expect({
+            method: 'post',
+            endpoint: '/api/checkin',
+            request: {
+                studentId: '123456',
+                itemAddress: 'iGwEZUvfA'
+            },
+            response
+        });
+        return checkIn('123456', 'iGwEZUvfA').then(data => {
+            assert.deepEqual(data, response.data);
+            mockServer.validate();
+        });
+    });
+
+    it('checkOutItems', () => {
+        let response = {
+            status: 'success'
+        };
+        mockServer.expect({
+            method: 'post',
+            endpoint: '/api/checkout',
+            request: {
+                studentId: '123456',
+                itemAddresses: ['iGwEZUvfA', 'iGwEZVHHE']
+            },
+            response
+        });
+        return checkOutItems('123456', ['iGwEZUvfA', 'iGwEZVHHE']).then(data => {
+            assert.isUndefined(data);
+            mockServer.validate();
+        });
+    });
+
+    it('checkOutItems (with code)', () => {
+        let response = {
+            status: 'success'
+        };
+        mockServer.expect({
+            method: 'post',
+            endpoint: '/api/checkout',
+            request: {
+                studentId: '123456',
+                itemAddresses: ['iGwEZUvfA', 'iGwEZVHHE'],
+                adminCode: 'abcdef'
+            },
+            response
+        });
+        return checkOutItems('123456', ['iGwEZUvfA', 'iGwEZVHHE'], 'abcdef').then(data => {
+            assert.deepEqual(data, response.data);
```

Comment:
Is there any particular reason for this assert? It is asserting undefined against undefined.